contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/spf13/viper"
+)
+
+type FlowInfoStatus int32
+
+const (
+	FlowInfoOneEntry   FlowInfoStatus = 1
+	FlowInfoTwoEntries FlowInfoStatus = 2
+	FlowInfoOldFlow    FlowInfoStatus = 3
+)
+
+// time for cleanup of old connections
+const timeoutPeriod = 20
+
+type accRejFlowInfo struct {
+	Status                   FlowInfoStatus
+	UUID1                    string
+	UUID2                    string
+	MetricReported           flow.FlowMetric
+	LastUpdateMetricReported flow.FlowMetric // FIXME - we might not need this
+	Saw1                     bool
+	Saw2                     bool
+	Flow1                    *flow.Flow
+	Flow2                    *flow.Flow
+}
+
+type accRejAction struct {
+	// table of saved flows
+	accRejState map[string]*accRejFlowInfo
+	timeOfLastCleanup	int64
+}
+
+// NewAction - creates data structures needed for vpc Accept/Reject based on measurements flows at 2 points
+func NewAction(cfg *viper.Viper) (interface{}, error) {
+	accRejState := make(map[string]*accRejFlowInfo)
+	now := time.Now()
+	secs := now.Unix()
+	accRej := &accRejAction{
+		accRejState:  accRejState,
+		timeOfLastCleanup: secs,
+	}
+	return accRej, nil
+}
+
+// initAction - initialize internal data structure for this round
+func (t *accRejAction) initAction() {
+	// reset the entries in the table for this round
+	for _, ars := range t.accRejState {
+		ars.Saw1 = false
```

Comment:
redundant it is already initialized to `false` by default

Reply:
We need to re-initialize this flag on each iteration.

Reply:
ok